salt/states/zpool.py

```diff
@@ -30,6 +30,23 @@
               /dev/disk2
               /dev/disk3
 
+    partitionpool:
+      zpool.present:
+        - config:
+            import: false
+            force: true
+        - properties:
+            comment: disk partition salty storage pool
+            ashift: '12'
+            feature@lz4_compress: enabled
+        - filesystem_properties:
+            compression: lz4
+            atime: on
+            relatime: on
+        - layout:
+            disk-0:
```

Comment:
Right that is used to filter mirror-N, raidz1-N,...

Try replacing it with:
````
if root_dev.count('-') != 1:
```

That should accept stuff like:
/dev/sda1
/dev/c1d1t1
/dev/disk/by-uuid/xxx-xxx-xxx-xxx

But still skip:
mirror-N
raidz1-N
disk-N
...